pkg/apis/m3dboperator/v1/pod_identity.go

```diff
@@ -26,20 +26,36 @@ type PodIdentitySource string
 const (
 	// PodIdentitySourcePodUID derives identity from UID of the pod.
 	PodIdentitySourcePodUID PodIdentitySource = "PodUID"
-	// PodIdentitySourcePodName derives identity from the name of the pod.
-	PodIdentitySourcePodName PodIdentitySource = "PodName"
+
+	// PodIdentitySourceNodeSpecExternalID derives identity from the 'externalID'
+	// spec field of the node. Note this field was deprecated after Kubernetes
+	// 1.11.
+	PodIdentitySourceNodeSpecExternalID PodIdentitySource = "NodeSpecExternalID"
+
+	// PodIdentitySourceNodeSpecProviderID derives identity from the 'providerID'
+	// spec field of the node.
+	PodIdentitySourceNodeSpecProviderID PodIdentitySource = "NodeSpecProviderID"
```

Comment:
what is the difference between External ID and Provider ID? Like, what do they actually mean in this context

Reply:
It's kind of weird, but provider ID is the node name but external ID is the unique underlying VM ID, iirc it's the `id` field in these docs: https://cloud.google.com/compute/docs/storing-retrieving-metadata#default

I think we have to clean this up another PR since the field becomes deprecated in 1.11